pkg/proto/nbcontract/v1/customsearchdomain.proto

```diff
@@ -0,0 +1,11 @@
+syntax = "proto3";
+package nbcontract.v1;
+
+message CustomSearchDomain {
```

Comment:
I don't know enough about this space, but find the "custom search domain" here a little confusing. Yes, looking at the script, it does add extra `dns-search` to the interface, but it looks like it also joins the node to the AD domain/realm, which seem a much more important/impactful action, that I would never have guessed from the name. Should we revisit the name? (Maybe tracking down the documented AKS/RP functionality or data model helps?)

Reply:
from AKS_RP datamodel, its the same https://msazure.visualstudio.com/CloudNativeCompute/_git/aks-rp?path=/protos/hcp/types/managedcluster/profiles/v1/linuxprofile.proto&version=GBmaster&line=33&lineEnd=34&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents

I wasnt able to find documentation in our wiki and the blame on this goes back to 3 years when AB was created. I can try digging up more.